pkg/chains/formats/slsa/v2alpha2/internal/resolved_dependencies/resolved_dependencies.go

```diff
@@ -116,6 +125,73 @@ func PipelineRun(ctx context.Context, pro *objects.PipelineRunObject) ([]v1.Reso
 	if err != nil {
 		return nil, err
 	}
+
+	// add on-cluster pipeline and tasks
+	rd, err := addLocalPipelineRunRefs(pro)
+	if err != nil {
+		return nil, err
+	}
+	resolvedDependencies = append(resolvedDependencies, rd...)
+	return resolvedDependencies, nil
+}
+
+// addLocalTaskRunRef adds the spec of on-cluster referenced task used by the taskrun
+func addLocalTaskRunRef(tro *objects.TaskRunObject) ([]v1.ResourceDescriptor, error) {
+	resolvedDependencies := []v1.ResourceDescriptor{}
+	tSpec := tro.Status.TaskSpec
+	if tSpec != nil {
+		if tro.Spec.TaskRef != nil && tro.Spec.TaskRef.Resolver == "" {
```

Comment:
Can't these conditions be merged to one?

Reply:
I think we can.